src/components/UserInfoTable.tsx

```diff
@@ -0,0 +1,221 @@
+  privilegeLevel: PrivilegeLevel;
+  phoneNumber: string;
+  profilePicture: string | null;
+  photoRelease: boolean;
+  detailsLink: string;
+}
+const TablePhotoConsent = (text: UserSummary, record: UserSummary) => (
+  <Space size="middle">{record.photoRelease ? 'Yes' : 'No'}</Space>
+);
+const TableActions = (text: UserSummary, record: UserSummary) => (
+  <Space size="middle">
+    <RouterLink to={record.detailsLink}>
+      <Link>View {record.firstName}'s Family Details</Link>
+    </RouterLink>
+  </Space>
+);
+
+interface UserInfoTableProps {
+  users: UserSummary[];
+}
+
+const UserInfoTable: React.FC<UserInfoTableProps> = ({ users }) => {
+  const [searchText, setSearchText] = useState<string>();
+  const [searchedColumn, setSearchedColumn] = useState<string>();
+
+  const getColumnSearchProps: (
+    dataIndex: string,
+  ) => {
+    filterDropdown: (props: {
+      setSelectedKeys: (value: string[]) => void;
+      selectedKeys: string[];
+      confirm: (props?: { closeDropdown: boolean }) => void;
+      clearFilters: () => void;
+    }) => JSX.Element;
+  } = (dataIndex: string) => ({
+    filterDropdown: ({
+      setSelectedKeys,
+      selectedKeys,
+      confirm,
+      clearFilters,
+    }) => (
+      <div style={{ padding: 8 }}>
+        <Input
+          placeholder={`Search ${dataIndex}`}
+          value={selectedKeys[0]}
+          onChange={(e) =>
+            setSelectedKeys(e.target.value ? [e.target.value] : [])
+          }
+          onPressEnter={() => handleSearch(selectedKeys, confirm, dataIndex)}
+          style={{ width: 188, marginBottom: 8, display: 'block' }}
+        />
+        <Space>
+          <Button
+            type="primary"
+            onClick={() => handleSearch(selectedKeys, confirm, dataIndex)}
+            icon={<SearchOutlined />}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Search
+          </Button>
+          <Button
+            onClick={() => handleReset(clearFilters)}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Reset
+          </Button>
+          <Button
+            type="link"
+            size="small"
+            onClick={() => {
+              confirm({ closeDropdown: false });
+              setSearchText(selectedKeys[0]);
+              setSearchedColumn(dataIndex);
+            }}
+          >
+            Filter
+          </Button>
+        </Space>
+      </div>
+    ),
+    filterIcon: (filtered: string) => (
+      <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
+    ),
+    onFilter: (value: string, record: { [key: string]: string }) =>
+      record[dataIndex]
+        ? record[dataIndex]
+            .toString()
+            .toLowerCase()
+            .includes(value.toLowerCase())
+        : '',
+    render: (text: string) =>
+      searchedColumn === dataIndex ? (
+        <Highlighter
+          highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
```

Comment:
What color is this? Consider defining a constant so the reader easily understands this.

Is the padding at default non-zero?

Reply:
Just stole this straight from the AntD docs, not super sure about the padding but I'll bring the color out